analyzers/src/SonarAnalyzer.CSharp/Rules/FieldShouldBeReadonly.cs

```diff
@@ -304,34 +306,45 @@ private void ProcessExpressionOnField(ExpressionSyntax expression, IFieldSymbol
 
                     if (!expression.RemoveParentheses().IsOnThis())
                     {
-                        this.readonlyFieldCollector.excludedFields.Add(fieldSymbol);
+                        readonlyFieldCollector.excludedFields.Add(fieldSymbol);
                         return;
                     }
 
-                    if (!(this.partialTypeDeclaration.SemanticModel.GetEnclosingSymbol(expression.SpanStart) is IMethodSymbol constructorSymbol))
+                    if (partialTypeDeclaration.SemanticModel.GetEnclosingSymbol(expression.SpanStart) is not IMethodSymbol methodSymbol)
                     {
-                        this.readonlyFieldCollector.excludedFields.Add(fieldSymbol);
+                        readonlyFieldCollector.excludedFields.Add(fieldSymbol);
                         return;
                     }
 
-                    if (constructorSymbol.MethodKind == MethodKind.Constructor &&
-                        constructorSymbol.ContainingType.Equals(fieldSymbol.ContainingType))
+                    if (methodSymbol.MethodKind == MethodKind.Constructor
+                        && methodSymbol.ContainingType.Equals(fieldSymbol.ContainingType))
                     {
-                        this.readonlyFieldCollector.assignedAsReadonly.Add(fieldSymbol);
+                        readonlyFieldCollector.assignedAsReadonly.Add(fieldSymbol);
                     }
                     else
                     {
-                        this.readonlyFieldCollector.excludedFields.Add(fieldSymbol);
+                        readonlyFieldCollector.excludedFields.Add(fieldSymbol);
                     }
                 }
 
-                private static bool IsFieldRelevant(IFieldSymbol fieldSymbol)
+                private static bool IsFieldRelevant(IFieldSymbol fieldSymbol) =>
+                    fieldSymbol is { IsStatic: false, IsConst: false, IsReadOnly: false, DeclaredAccessibility: Accessibility.Private };
+
+                private static IEnumerable<ExpressionSyntax> IdentifierNames(ExpressionSyntax expression)
```

Comment:
this is not really `IdentifierNames`, right? it's returning expressions, not IdentifierNameSyntax ...

could you please find a better name? what does this method do? It either gets the expressions inside a tuple, or it returns the expression

`TupleExpressionsOrSelf` sounds more like what it does, right?